libsolidity/analysis/NameAndTypeResolver.cpp

```diff
@@ -132,76 +128,72 @@ bool NameAndTypeResolver::performImports(SourceUnit& _sourceUnit, map<string, So
 	return !error;
 }
 
-bool NameAndTypeResolver::resolveNamesAndTypes(ContractDefinition& _contract)
+bool NameAndTypeResolver::resolveNamesAndTypes(ASTNode& _node, bool _resolveInsideCode)
 {
+	bool success = true;
 	try
 	{
-		m_currentScope = m_scopes[_contract.scope()].get();
-		solAssert(!!m_currentScope, "");
+		if (ContractDefinition* contract = dynamic_cast<ContractDefinition*>(&_node))
+		{
+			m_currentScope = m_scopes[contract->scope()].get();
+			solAssert(!!m_currentScope, "");
 
-		ReferencesResolver resolver(m_errors, *this, nullptr);
-		bool success = true;
-		for (ASTPointer<InheritanceSpecifier> const& baseContract: _contract.baseContracts())
-			if (!resolver.resolve(*baseContract))
-				success = false;
+			for (ASTPointer<InheritanceSpecifier> const& baseContract: contract->baseContracts())
+				if (!resolveNamesAndTypes(*baseContract, false))
```

Comment:
If you want to keep the old behavior, `!resolveNamesAndTypes(*baseContract, true)` looks more similar.